test-framework/core/src/main/java/org/keycloak/test/framework/realm/ManagedRealmCleanup.java

```diff
@@ -0,0 +1,49 @@
+package org.keycloak.test.framework.realm;
+
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.representations.idm.RealmRepresentation;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class ManagedRealmCleanup {
+
+    private final List<RealmCleanup> cleanupTasks = new LinkedList<>();
+
+    public ManagedRealmCleanup add(RealmCleanup realmCleanup) {
+        this.cleanupTasks.add(realmCleanup);
+        return this;
+    }
+
+    public ManagedRealmCleanup deleteUsers() {
+        return add(r -> r.users().list().forEach(u -> r.users().delete(u.getId()).close()));
+    }
+
+    void resetToOriginalRepresentation(RealmRepresentation rep) {
```

Comment:
Is it intentional to not using public?
```suggestion
    public void resetToOriginalRepresentation(RealmRepresentation rep) {
```

Reply:
Yes, this should only be called by `ManagedRealm`